### PR TITLE
add test to cover stress.py fails

### PR DESCRIPTION
**Pull request by pmnoxx**
Add a new test, which should. detect failures fixed in https://github.com/nearprotocol/nearcore/pull/3036

Adds test for https://github.com/nearprotocol/nearcore/pull/3036

### CODE REVIEW

File: pytest/tests/sanity/proxy_loose_block_messages.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                first_block_time.value = time.time()
+            h = msg.Block.BlockV1.header.BlockHeaderV2.inner_lite.height
+            if h not in height2producer:
+                height2producer[h] = fr
+                max_height.value = h
+                print("%.2f new height at %d" %  (time.time() - first_block_time.value, h))
+                current_block_producer.value = fr
+            print(h, fr, to,  msg, msg.Block.enum)
+
+            if height2producer[h] == NODES - 1 and fr == NODES - 1 and to < (NODES - 1) / 2:
```
Comment (SkidanovAlex) [inline pytest/tests/sanity/proxy_loose_block_messages.py:39]: We want to block the first occurrence of `(h, fr, to)`.

File: pytest/xx.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from multiprocessing import Process, Value, Array
```
Comment (SkidanovAlex) [inline pytest/xx.py:1]: was this intended to be added to the PR?

File: pytest/tests/sanity/proxy_loose_block_messages.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        super().__init__(*args, **kwargs)
+        self.peers_response = 0
+        self.handler_state = handler_state
+
+    async def handle(self, msg, fr, to):
+        global height2producer, first_block_time
+
+        if msg.enum == 'Block':
+            h = msg.Block.BlockV1.header.BlockHeaderV2.inner_lite.height
+            if h > self.handler_state.value:
```
Comment (SkidanovAlex) [inline pytest/tests/sanity/proxy_loose_block_messages.py:32]: is `handler_state.value` used anywhere?

File: pytest/tests/sanity/proxy_loose_block_messages.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                self.handler_state.value = h
+            print(h, fr, to,  msg, msg.Block.enum)
+            tpl = (h, fr, to)
+
+            if h % NODES == NODES - 1 and tpl not in ignored_messages:
```
Comment (SkidanovAlex) [inline pytest/tests/sanity/proxy_loose_block_messages.py:37]: This drops it to all the recipients, not just half of them, right?

**Comment by bowenwang1996** [comment]
@pmnoxx @SkidanovAlex what is the status of this PR?

**Comment by pmnoxx** [comment]
https://nayduck.near.org/#/run/2202

File: pytest/tests/sanity/proxy_loose_block_messages.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+from proxy import ProxyHandler
+
+from multiprocessing import Value
+
+TIMEOUT = 120
+NODES = 9
+ignored_messages = set()
+max_height = Value('i', 0)
+
+
```
Comment (mm-near) [inline pytest/tests/sanity/proxy_loose_block_messages.py:17]: can you explain what this test does?

**Review by mm-near — CHANGES_REQUESTED** [review]

**Pull request closed without merging**